File: weekly/327/q1.cc
class Solution {
public:
    int maximumCount(vector<int>& nums) {
        return max(count_if(nums.begin(), nums.end(), [](const int& x) { return x > 0; }), count_if(nums.begin(), nums.end(), [](const int& x) { return x < 0; }));
    }
};

File: weekly/352_/q1.go
func longestAlternatingSubarray(nums []int, th int) (ans int) {
    cons := 0
    for _, a := range nums {
        if cons != 0 {
            if cons % 2 == 0 && a % 2 == 0 && a <= th {
                cons++
                continue
            }
            if cons % 2 == 1 && a % 2 == 1 && a <= th {
                cons++
                continue
            }
            ans = max(ans, cons)
            cons = 0
        }

        if cons == 0 && a % 2 == 0 && a <= th {
            cons++
            continue
        }
    }

    if cons != 0 {
        ans = max(ans, cons)
    }

    return
}

func max(a, b int) int {
    if a > b {
        return a
    } else {
        return b
    }
}

